### PR TITLE
[RFR] Winning condition

### PR DESCRIPTION
**Pull request by JulienMattiussi**
https://trello.com/c/GOa9GVKk
https://trello.com/c/AOxtNdDS

Logic a winning/ draw is in place
Writing tests on that in progress

### CODE REVIEW

File: src/game/state.py [hunk 1]
```diff
@@ -20,12 +21,13 @@ def from_dictionary(self, initial_state_dictionary):
                 self.place_piece(key, val)
```
Comment (jdemangeon) [inline src/game/state.py:21]: This function name (from_dictionary) suppose that you get a new state from dictionary. Please return a new State from it or rename it

File: src/game/state.py [hunk 2]
```diff
@@ -62,3 +65,41 @@ def check_position_availability(self, x, y):
 
     def switch_player(self):
         self.game_turn.player_one_active = not self.game_turn.player_one_active
+
+    def check_winner(self):
+        if self.check_raws_winning() or self.check_columns_winning() or self.check_diags_winning():
+            if self.game_turn.player_one_active:
+                player_name = "Player 1"
+            else:
```
Comment (jdemangeon) [inline src/game/state.py:73]: Useless else

File: src/game/state.py [hunk 3]
```diff
@@ -62,3 +65,41 @@ def check_position_availability(self, x, y):
...
+        return False
+
+    def check_columns_winning(self):
+        for i in range(3):
+            if Piece.check_line_winning(self.grid[0][i], self.grid[1][i], self.grid[2][i], self.grid[3][i]):
+                return True
+        return False
+
+    def check_diags_winning(self):
+        if Piece.check_line_winning(self.grid[0][0], self.grid[1][1], self.grid[2][2], self.grid[3][3]):
```
Comment (jdemangeon) [inline src/game/state.py]: Use "or" instead of elif

return cond1 or cond2

File: src/game/state.py [hunk 4]
```diff
@@ -98,3 +100,6 @@ def check_diags_winning(self):
         elif Piece.check_line_winning(self.grid[0][3], self.grid[1][2], self.grid[2][1], self.grid[3][0]):
             return True
         return False
+
+    def is_selected_piece(self):
```
Comment (Kmaschta) [inline src/game/state.py]: You can turn this method into an attribute with the `@property` decorator

File: src/game/state.py (hunk 2 above)
Comment (JulienMattiussi) [reply src/game/state.py:73]: If I remove the "else", I will have a useless double allocation 1 time /2 when setting the "player 1" name
I think I perfer this righting for now

**Review by jdemangeon — APPROVED** [review]

**Pull request merged**